opnode/node/driver.go

```diff
@@ -0,0 +1,87 @@
+package node
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/holiman/uint256"
+)
+
+type InputItem struct {
+	Block    *types.Block
+	Receipts []*types.Receipt
+}
+
+func (item *InputItem) Epoch() uint64 {
+	return item.Block.NumberU64()
+}
+
+// sanity check that the receipts are consistent with the block data
+func (item *InputItem) CheckReceipts() bool {
+	hasher := trie.NewStackTrie(nil)
+	computed := types.DeriveSha(types.Receipts(item.Receipts), hasher)
+	return item.Block.ReceiptHash() == computed
+}
+
+type DepositTransaction struct {
+	Sender common.Address
+	// TODO some flag/bool for contract-deploy transaction. Or it can be *To with nil
+	Flags uint8
+	To    common.Address
+	Value uint256.Int
+	Data  []byte
+}
+
+func (dep *DepositTransaction) Decode(data []byte) (valid bool) {
+	// TODO
+	return false
+}
+
+type OpaqueTransaction []byte
+
+type OutputItem struct {
+	// TODO: convert to L1 attributes deposit
+	L1BlockHash   common.Hash
+	L1BlockNumber uint64
+
+	// To be converted to full block with Engine API
+	UserDepositTxs []*DepositTransaction
+}
+
+var DepositGatewayAddr = common.HexToAddress("0xaaaaaaaaaTODO")
```

Comment:
What if we changed this to `DepositContractAddr`?

Reply:
I think there are contracts named like ETH and token gateway, but it's really just a placeholder. Let's address naming in the block-derivation work.